java/flight/flight-core/src/test/java/org/apache/arrow/flight/TestCallOptions.java

```diff
@@ -64,14 +66,29 @@ public void underTimeout() {
     });
   }
 
-  void test(Consumer<FlightClient> testFn) {
+  @Test
+  public void propertyHeader() {
```

Comment:
Let's add more tests with multiple properties, properties with special chars in name/value.